tests/rptest/wasm/native_kafka_consumer.py

```diff
@@ -57,19 +60,29 @@ def _finished_consume(self):
         return True
 
     def _run(self):
-        def stop_consume(empty_attempts):
-            waited_enough = empty_attempts >= self._max_attempts
-            return self.is_finished() or self._finished_consume(
-            ) or waited_enough
-
         consumer = self._init_consumer()
         empty_reads = 0
-        while not stop_consume(empty_reads):
+        empty_reads_post_complete = 0
+        while True:
+            if self.is_finished():
+                break  # User stopped background task
+            if self._finished_consume():
+                # The idea is to not stop consuming even if the bounds
+                # have been reached, and stop when there really is not more data
+                empty_reads_post_complete += 1
+                if empty_reads_post_complete >= 3:
```

Comment:
I don't see a sleep/wait in the _finished_consume path, so in this case the loop is probably going to spin 3 times very quickly.  This code reads like it's meant to be a retry, so you probably meant to sleep before the next try?

Reply:
Theres a timeout passed to poll which is 1s